main.go

```diff
@@ -47,166 +46,34 @@ const (
-	}
-
-	if err := input.ValidateWithOptions(configs.UseDeprecatedExport, "yes", "no"); err != nil {
-		return errors.New("issue with input UseDeprecatedExport: " + err.Error())
-	}
-	if configs.CustomExportOptionsPlistContent != "" {
-		var options map[string]interface{}
-		if _, err := plist.Unmarshal([]byte(configs.CustomExportOptionsPlistContent), &options); err != nil {
-			return errors.New("issue with input CustomExportOptionsPlistContent: " + err.Error())
-		}
-	}
-
-	// xcodebuild configs
-	if err := input.ValidateWithOptions(configs.OutputTool, "xcpretty", "xcodebuild"); err != nil {
-		return errors.New("issue with input OutputTool: " + err.Error())
-	}
-
-	if configs.Workdir != "" {
-		if err := input.ValidateIfDirExists(configs.Workdir); err != nil {
-			return errors.New("issue with input Workdir: " + err.Error())
-		}
-	}
-	if err := input.ValidateIfPathExists(configs.ProjectPath); err != nil {
-		return errors.New("issue with input ProjectPath: " + err.Error())
-	}
-	if err := input.ValidateIfNotEmpty(configs.Scheme); err != nil {
-		return errors.New("issue with input Scheme: " + err.Error())
-	}
-	if err := input.ValidateIfNotEmpty(configs.OutputDir); err != nil {
-		return errors.New("issue with input OutputDir: " + err.Error())
-	}
-	if err := input.ValidateWithOptions(configs.IsCleanBuild, "yes", "no"); err != nil {
-		return errors.New("issue with input IsCleanBuild: " + err.Error())
-	}
-
-	// step output configs
-	if err := input.ValidateWithOptions(configs.IsExportXcarchiveZip, "yes", "no"); err != nil {
-		return errors.New("issue with input IsExportXcarchiveZip: " + err.Error())
-	}
-	if err := input.ValidateWithOptions(configs.ExportAllDsyms, "yes", "no"); err != nil {
-		return errors.New("issue with input ExportAllDsyms: " + err.Error())
-	}
-	if err := input.ValidateWithOptions(configs.VerboseLog, "yes", "no"); err != nil {
-		return errors.New("issue with input VerboseLog: " + err.Error())
-	}
-
-	return nil
+// configs ...
+type configs struct {
+	ExportMethod               string `env:"export_method,opt[auto-detect,app-store,ad-hoc,enterprise,development]"`
+	UploadBitcode              string `env:"upload_bitcode,opt[yes,no]"`
+	CompileBitcode             string `env:"compile_bitcode,opt[yes,no]"`
+	ICloudContainerEnvironment string `env:"icloud_container_environment"`
+	TeamID                     string `env:"team_id"`
+
+	UseDeprecatedExport               string `env:"use_deprecated_export,opt[yes,no]"`
+	ForceTeamID                       string `env:"force_team_id"`
+	ForceProvisioningProfileSpecifier string `env:"force_provisioning_profile_specifier"`
+	ForceProvisioningProfile          string `env:"force_provisioning_profile"`
+	ForceCodeSignIdentity             string `env:"force_code_sign_identity"`
+	CustomExportOptionsPlistContent   string `env:"custom_export_options_plist_content"`
+
+	OutputTool        string `env:"output_tool,opt[xcpretty,xcodebuild]"`
+	Workdir           string `env:"workdir,dir"`
```

Comment:
workdir is not required input

Reply:
Fixed in:
https://github.com/bitrise-steplib/steps-xcode-archive/pull/151/commits/1978f1e243a3c3ece84cb3b785a70bb85b505e37